Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesPanel.java

```diff
@@ -0,0 +1,592 @@
+
+            private static final String SELECT_DATA_SOURCES_IMAGE = "select obj_id, name from tsk_image_names where obj_id in (SELECT obj_id FROM tsk_objects WHERE obj_id in (select obj_id from data_source_info))";
+
+            private void updateUi() {
+
+                String[] dataSourcesNames = new String[CommonFilesPanel.this.dataSourceMap.size()];
+
+                //only enable all this stuff if we actually have datasources
+                if (dataSourcesNames.length > 0) {
+                    dataSourcesNames = CommonFilesPanel.this.dataSourceMap.values().toArray(dataSourcesNames);
+                    CommonFilesPanel.this.dataSourcesList = new DataSourceComboBoxModel(dataSourcesNames);
+                    CommonFilesPanel.this.selectDataSourceComboBox.setModel(CommonFilesPanel.this.dataSourcesList);
+
+                    boolean multipleDataSources = this.caseHasMultipleSources();
+                    CommonFilesPanel.this.allDataSourcesRadioButton.setEnabled(multipleDataSources);
+                    CommonFilesPanel.this.allDataSourcesRadioButton.setSelected(multipleDataSources);
+
+                    if (!multipleDataSources) {
+                        CommonFilesPanel.this.withinDataSourceRadioButton.setSelected(true);
+                        withinDataSourceSelected(true);
+                    }
+
+                    CommonFilesPanel.this.searchButton.setEnabled(true);
+                } else {
+                    MessageNotifyUtil.Message.info(Bundle.CommonFilesPanel_buildDataSourcesMap_updateUi_noDataSources());
+                    CommonFilesPanel.this.cancelButtonActionPerformed(null);
+                }
+            }
+
+            private boolean caseHasMultipleSources() {
+                return CommonFilesPanel.this.dataSourceMap.size() >= 2;
+            }
+
+            private void loadLogicalSources(SleuthkitCase tskDb, Map<Long, String> dataSouceMap) throws TskCoreException, SQLException {
+                //try block releases resources - exceptions are handled in done()
+                try (CaseDbQuery query = tskDb.executeQuery(SELECT_DATA_SOURCES_LOGICAL)) {
+                    ResultSet resultSet = query.getResultSet();
+                    while (resultSet.next()) {
+                        Long objectId = resultSet.getLong(1);
+                        String dataSourceName = resultSet.getString(2);
+                        dataSouceMap.put(objectId, dataSourceName);
+                    }
+                }
+            }
+
+            private void loadImageSources(SleuthkitCase tskDb, Map<Long, String> dataSouceMap) throws SQLException, TskCoreException {
+                //try block releases resources - exceptions are handled in done()
+                try (CaseDbQuery query = tskDb.executeQuery(SELECT_DATA_SOURCES_IMAGE)) {
+                    ResultSet resultSet = query.getResultSet();
```

Comment:
I think I missed this one - please move the ResultSet into the try with resources

Reply:
I think you caught one, but I misused this paradigm in the same exact way twice.